config/routes.rb

```diff
@@ -588,6 +588,10 @@
       resources :media, only: [:create]
       get '/search', to: 'search#index', as: :search
       resources :suggestions, only: [:index]
+      resources :filters,     only: [:index, :create, :show, :update, :destroy] do
+        resources :keywords, only: [:index, :create], controller: 'filters/keywords'
+      end
+      resources :filter_keywords, only: [:show, :update, :destroy], controller: 'filters/keywords'
```

Comment:
I'm not entirely sure why this exists at the top level as well as under the filters resource. I would prefer to avoid mixing namespacing schemes (`filter_keywords` and `filters/:id/keywords`).

Reply:
Those are recommendations from https://guides.rubyonrails.org/routing.html#limits-to-nesting

Reply:
Right, but most Rails apps are simple CRUD and don't really care how URLs look, and aren't stuck with them for years to come. You could try doing something like:

```ruby
namespace :filters do
  resources :keywords, only: [:show, :update, :destroy], controller: 'filters/keywords'
end
```

So the result would be something like:

```
GET  /api/v1/filters/:id/keys
POST /api/v1/filters/:id/keys

GET    /api/v1/filters/keys/:id
PUT    /api/v1/filters/keys/:id
DELETE /api/v1/filters/keys/:id
```

Reply:
Moved `/api/v2/filter_keywords/` to `/api/v1/filters/keywords/`.